test_bin/src/main.rs

```diff
@@ -61,70 +66,134 @@ lazy_static! {
     ];
 }
 
+#[cfg_attr(tarpaulin, skip)]
 fn print_three_nodes_test_name(print_str: &str, test_fn: ThreeNodesTestFn) {
     print_test_name(print_str, test_fn as *mut std::os::raw::c_void);
 }
 
+#[cfg_attr(tarpaulin, skip)]
 fn print_two_nodes_test_name(print_str: &str, test_fn: TwoNodesTestFn) {
     print_test_name(print_str, test_fn as *mut std::os::raw::c_void);
 }
 
 /// Print name of test function
+#[cfg_attr(tarpaulin, skip)]
 fn print_test_name(print_str: &str, test_fn: *mut std::os::raw::c_void) {
     backtrace::resolve(test_fn, |symbol| {
         let mut full_name = symbol.name().unwrap().as_str().unwrap().to_string();
         let mut test_name = full_name.split_off("holochain_test_bin::".to_string().len());
         test_name.push_str("()");
-        println!("{}{}", print_str, test_name);
+        log_i!("{}{}", print_str, test_name);
     });
 }
 
 // this is all debug code, no need to track code test coverage
 #[cfg_attr(tarpaulin, skip)]
-fn usage() {
-    println!("Usage: holochain_test_bin <path_to_n3h>");
-    std::process::exit(1);
+fn load_config_file(filepath: &str) -> serde_json::Value {
+    let config_file =
+        File::open(filepath).expect("Failed to open filepath on Network Test config.");
+    serde_json::from_reader(config_file).expect("file is not proper JSON")
 }
 
 // this is all debug code, no need to track code test coverage
 #[cfg_attr(tarpaulin, skip)]
 fn main() {
-    // Check args
+    // Check args ; get config filepath
     let args: Vec<String> = std::env::args().collect();
+    let mut config_path = String::new();
     if args.len() != 2 {
-        usage();
+        println!(
+            "Usage: No config file supplied. Using default config: test_bin/data/test_config.json"
+        );
+    } else {
+        config_path = args[1].clone();
     }
-    let n3h_path = args[1].clone();
-    if n3h_path == "" {
-        usage();
+    if config_path == "" {
+        println!(
+            "Usage: No config file supplied. Using default config: test_bin/data/test_config.json"
```

Comment:
looks like we might end up printing this `Usage` line twice with the `args.len()` check above.

Reply:
Good call. Indeed I got a bit sloppy there.
Fixed.